extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java

```diff
@@ -259,4 +260,36 @@ private static void wrapExceptions(Runnable runnable)
       return null;
     });
   }
+
+  @VisibleForTesting
+  private KafkaConsumer<byte[], byte[]> getKafkaConsumer(Map<String, Object> consumerConfigs)
+  {
+    final Properties props = new Properties();
+    addConsumerPropertiesFromConfig(props, sortingMapper, consumerProperties);
+    props.putAll(consumerConfigs);
+
+    ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+      Deserializer keyDeserializerObject = getKafkaDeserializer(props, "key.deserializer");
+      Deserializer valueDeserializerObject = getKafkaDeserializer(props, "value.deserializer");
+
+      return new KafkaConsumer<>(props, keyDeserializerObject, valueDeserializerObject);
+    }
+    finally {
+      Thread.currentThread().setContextClassLoader(currCtxCl);
+    }
+  }
+
+  @VisibleForTesting
+  public KafkaRecordSupplier(
+      Map<String, Object> consumerProperties,
+      ObjectMapper sortingMapper,
+      Map<String, Object> consumerConfigs
+  )
+  {
+    this.consumerProperties = consumerProperties;
+    this.sortingMapper = sortingMapper;
+    this.consumer = getKafkaConsumer(consumerConfigs);
+  }
```

Comment:
Instead of adding a constructor that adds a `Map<String, Object> consumerConfigs` parameter, how about making it:
```
@VisibleForTesting
public KafkaRecordSupplier(
    Map<String, Object> consumerProperties,
    ObjectMapper sortingMapper,
    KafkaConsumer<byte[], byte[]> consumer
)
```

You'd move this new constructor to the top of this file below the original one and then have the original one delegate to your new one by passing in the existing `getKafkaConsumer()`. In `KafkaSupervisorTest`, for your regression test, you'd then use your new `@VisibleForTesting` constructor and move the method you added for creating a `KafkaConsumer` there since it's only intended for tests.

Reply:
getKafkaConsumer() is a private method.

Reply:
and `getKafkaConsumer ` depends on another private method,  `addConsumerPropertiesFromConfig`, copy both method to test file?